### PR TITLE
fix/AUT-2551/fix-response-saving-auth-math-entry-pci

### PR DESCRIPTION
**Pull request by marpesia**
**Related to:** [AUT-2551](https://oat-sa.atlassian.net/browse/AUT-2551)

**Description:**
MathEntry Interaction was not able to save a correct response on authoring.
The problem is on a condition that is checking a boolean value, but the property has a default 'false' string value instead of a false boolean value. I tried to change the default value, but it has unexpected consequences, so the best approach and safest is to add an OR condition with this 'false' string, so both cases scenarios are in.
There are 2 scenarios, the first time the item is created with the PCI, and the second is when you edit an existing item with the PCI. In each scenario, the value type is different.

**Changes:**

- Add OR condition on the change listener function.

**How to check:**

- Check capacity to save a correct response on the item creation with a MathEntry Interaction PCI.
- Check capacity to save a correct response on the item edition with a MathEntry Interaction PCI.
- Check data on the `saveItem` XML on saving the item. It must contain the correct result
image
- Check output on Preview.
image

TAE environment on the ticket
https://oat-sa.atlassian.net/browse/AUT-2551?focusedCommentId=198672

### CODE REVIEW

**Review by bziondik — COMMENTED** [review]
- [ ]  New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ]  New code is respecting code style rules
- [ ]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Review by bziondik — APPROVED** [review]

File: views/js/pciCreator/ims/mathEntryInteraction/creator/widget/states/Map.js [hunk 1]
```diff
@@ -130,7 +130,7 @@ define([
             interaction = self.widget.element;
 
         interaction.onPci('responseChange', function (latex) {
-            if (self.inGapMode(self) === false && self.activeEditId !== null) {
+            if ((self.inGapMode(self) === false || self.inGapMode(self) === 'false') && self.activeEditId !== null) {
```
Comment (jsconan) [inline views/js/pciCreator/ims/mathEntryInteraction/creator/widget/states/Map.js:133]: **question:** Why not fix the `inGapMode` method instead?

https://github.com/oat-sa/extension-tao-itemqti-pci/blob/51e90a24a297bb3881bfb302ce0ee06d268b8bc0/views/js/pciCreator/ims/mathEntryInteraction/creator/widget/states/Map.js#L309

**Review by jsconan — CHANGES_REQUESTED** [review]
Ok, this fix the issue, but it only patches the use not the source issue. 

I'd rather recommend fixing the API to respect the contract. A boolean is expected, so we need to return a boolean

File: views/js/pciCreator/ims/mathEntryInteraction/creator/widget/states/Map.js (hunk 1 above)
Comment (jsconan) [reply views/js/pciCreator/ims/mathEntryInteraction/creator/widget/states/Map.js:133]: **suggestion:** you could rather do something like this:

```javascript
    MathEntryInteractionStateResponse.prototype.inGapMode = function inGapMode() {
        var interaction = this.widget.element;
        var useGapExpression = interaction.prop('useGapExpression');
        return useGapExpression && useGapExpression !== 'false' || false;
    }
```

File: views/js/pciCreator/ims/mathEntryInteraction/creator/widget/states/Map.js [hunk 2]
```diff
@@ -130,7 +130,7 @@ define([
             interaction = self.widget.element;
 
         interaction.onPci('responseChange', function (latex) {
-            if (self.inGapMode(self) === false && self.activeEditId !== null) {
+            if ((self.inGapMode(self) === false) && self.activeEditId !== null) {
```
Comment (jsconan) [inline views/js/pciCreator/ims/mathEntryInteraction/creator/widget/states/Map.js:133]: **chore:** The additional parenthesis are useless and could have been removed...

**Review by jsconan — APPROVED** [review]
Looks good!

image

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] ~~New code is not subject to concurrency issues (if applicable)~~
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic

**Pull request merged**